bn-apps/memberships-management/membership-service/src/main/kotlin/net/corda/businessnetworks/membership/member/support/BusinessNetworkAwareInitiatedFlow.kt

```diff
@@ -33,10 +34,12 @@ abstract class BusinessNetworkAwareInitiatedFlow<out T>(protected val session: F
     @Suspendable
     private fun verifyMembership(initiator : Party) {
         val membership = subFlow(GetMembershipsFlow(bnoIdentity()))[initiator]
-        // inactive members can't transact on the network
-        if(membership == null || !membership.state.data.isActive()) {
+        val configuration = serviceHub.cordaService(MemberConfigurationService::class.java)
+        // 1. membership should exist
+        // 2. membership have to be active
+        // 3. membership have to be validated by the correct contract
+        if(membership == null || !membership.state.data.isActive() || membership.state.contract != configuration.membershipContractName()) {
             throw NotAMemberException(initiator)
```

Comment:
I think we need a different exception for all these three cases. It will cause a lot of head-scratching if we just throw NotAMemberException. Also do we need to check the contract? If the BNO say a member is a member then that should be authoritative enough.